llvm-bazel/llvm-project-overlay/mlir/BUILD

```diff
@@ -1,12 +1,8 @@
-# This file is licensed under the Apache License v2.0 with LLVM Exceptions.
-# See https://llvm.org/LICENSE.txt for license information.
-# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
```

Comment:
Why are we losing this in the file? Just seems inconsistent... But maybe I don't understand the context here.

Reply:
I've yet to really get a clear answer on what license headers are actually supposed to *do* and where they're required and how hard we should try to put them there: e.g. all of TF doesn't put license headers on its build files. Similarly, LLVM doesn't put license headers on its CMakeLists.txt files. In this case, the internal file doesn't have a header and it's exported to TF as well, so I can't really put an LLVM header on it. I could add the header on export, but it didn't seem worth the trouble.

For consistency, we could drop the header from the llvm subproject build files as well for now.

Reply:
I think we should add the header if we want this to eventually be viable for inclusion in the larger LLVM project FWIW.

I don't have any strong opinions about whether to do so in the TF project. But unless it is a significant burden, I think having the headers is useful documentation to have at the top of each file and I'd prefer to be consistent about it.

Reply:
Ok I added back the headers and I'll look at adding them on export. It shouldn't be too bad here because the header doesn't include the year, which is supposed to be the oldest creation date, so can't just be added as the current date each time.

I'm surprised you think this would cause issues for inclusion in the LLVM project though. The whole repo is licensed under the Apache-LLVM license and these headers are just reminders AFAIU. There aren't license headers in the CMakeLists.txt files in the llvm-project, which seems like the closest analogue.